sklearn/utils/validation.py

```diff
@@ -239,19 +240,30 @@ def _ensure_sparse_format(spmatrix, accept_sparse, dtype, copy,
     spmatrix_converted : scipy sparse matrix.
         Matrix that is ensured to have an allowed type.
     """
-    if accept_sparse in [None, False]:
-        raise TypeError('A sparse matrix was passed, but dense '
-                        'data is required. Use X.toarray() to '
-                        'convert to a dense numpy array.')
     if dtype is None:
         dtype = spmatrix.dtype
 
     changed_format = False
-    if (isinstance(accept_sparse, (list, tuple))
-            and spmatrix.format not in accept_sparse):
-        # create new with correct sparse
-        spmatrix = spmatrix.asformat(accept_sparse[0])
-        changed_format = True
+
+    if isinstance(accept_sparse, str):
+        accept_sparse = [accept_sparse]
+
+    if accept_sparse in [None, False]:
+        raise TypeError('A sparse matrix was passed, but dense '
+                        'data is required. Use X.toarray() to '
+                        'convert to a dense numpy array.')
+    elif (isinstance(accept_sparse, (list, tuple)) and
+          len(accept_sparse) and
+          isinstance(accept_sparse[0], str)):
+        # ensure correct sparse format
+        if spmatrix.format not in accept_sparse:
+            # create new with correct sparse
+            spmatrix = spmatrix.asformat(accept_sparse[0])
+            changed_format = True
+    elif accept_sparse is not True:
+        # any other type
+        raise ValueError('The parameter \'accept_sparse\' was '
```

Comment:
I'm trying to provide the value with `"accept_sparse=%s" % accept_sparse` but when an empty tuple is passed in (essentially `"%s" % ()"`), this errors with "not enough arguments for format string". Do you have a suggestion for how to format properly in case its just an empty tuple, aside from just checking for that specific case?